### PR TITLE
add Lambda service model

### PR DESCRIPTION
**Pull request by richardhboyd**
*Issue #, if available:* #65

*Description of changes:*
Added Lambda service model and a list-functions example. The example hasn't been tested.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/sdk/examples/lambda-list-functions/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            .with_env_filter("info")
+            .with_span_events(FmtSpan::CLOSE)
+            .init();
+    }
+
+    let config = Config::builder().region(region).build();
+
+    let client = Client::from_conf(config);
+
+    match client.list_functions().send().await {
```
[the remark this answers is not in the document]
Comment (rcoh) [reply aws/sdk/examples/lambda-list-functions/src/main.rs:49]: ah that makes sense. I mostly just care that the requests are being serialized / deserialized properly, if you don't use it in an example just knowing that you ran it and it worked is more or less good enough for me.

It also isn't crucial that these examples are runnable, people will more use them to copy-paste code out

File: aws/sdk/examples/lambda-list-functions/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+#[tokio::main]
+async fn main() {
+    let Opt { region, verbose } = Opt::from_args();
+
+    let region = EnvironmentProvider::new()
```
[the remark this answers is not in the document]
Comment (richardhboyd) [reply aws/sdk/examples/lambda-list-functions/src/main.rs]: done, I hope?

File: aws/sdk/examples/lambda-list-functions/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (rcoh) [reply aws/sdk/examples/lambda-list-functions/src/main.rs:34]: Thanks! Will debug

**Review by rcoh — APPROVED** [review]
LGTM thanks!

**Pull request merged**